src/System.Private.CoreLib/src/System/StubHelpers.cs

```diff
@@ -573,26 +635,45 @@ internal static class ObjectMarshaler
 
 #endif // FEATURE_COMINTEROP
 
-    internal static class ValueClassMarshaler
+    internal sealed class HandleMarshaler
     {
-        [MethodImpl(MethodImplOptions.InternalCall)]
-        internal static extern void ConvertToNative(IntPtr dst, IntPtr src, IntPtr pMT, ref CleanupWorkListElement pCleanupWorkList);
+        internal static unsafe IntPtr ConvertSafeHandleToNative(SafeHandle? handle, bool hasCleanupWorkList, ref CleanupWorkListElement? cleanupWorkList)
+        {
+            if (!hasCleanupWorkList)
```

Comment:
We don't have the `ldstr` instruction implemented in the native IL stub emit infrastructure. I felt it would just be easier to throw from within C# than to implement the instruction along with managing the string lifetimes.